src/server/scripts/Shadowlands/SanctumOfDomination/boss_sylvanas_windrunner.cpp

```diff
@@ -0,0 +1,8620 @@
+
+        _arrowAreaTriggerGUID = AreaTrigger::CreateNewMovementForceId(caster->GetMap(), DATA_AREATRIGGER_DOMINATION_ARROW);
+
+        target->ApplyMovementForce(_arrowAreaTriggerGUID, caster->GetPosition(), 3.20000004768371582f, MovementForceType::Gravity);
+    }
+
+    void AfterRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        Unit* target = GetTarget();
+
+        target->RemoveAurasDueToSpell(SPELL_DOMINATION_CHAIN_PERIODIC);
+
+        target->RemoveMovementForce(_arrowAreaTriggerGUID);
+    }
+
+    void Register() override
+    {
+        OnEffectApply += AuraEffectApplyFn(spell_sylvanas_windrunner_domination_chain::OnApply, EFFECT_0, SPELL_AURA_PERIODIC_DUMMY, AURA_EFFECT_HANDLE_REAL);
+        AfterEffectRemove += AuraEffectRemoveFn(spell_sylvanas_windrunner_domination_chain::AfterRemove, EFFECT_0, SPELL_AURA_PERIODIC_DUMMY, AURA_EFFECT_HANDLE_REAL);
+    }
+
+private:
+    ObjectGuid _playerGUID;
+    ObjectGuid _arrowAreaTriggerGUID;
+};
+
+// 349458 - Domination Chain (Periodic)
+class spell_sylvanas_windrunner_domination_chain_periodic : public AuraScript
+{
+    PrepareAuraScript(spell_sylvanas_windrunner_domination_chain_periodic);
+
+    void HandleEffectPeriodic(AuraEffect const* aurEff)
+    {
+        Unit* target = GetTarget();
+        Unit* caster = GetCaster();
+        if (!caster)
+            return;
+
+        if (aurEff->GetTickNumber() == 3)
+            caster->CastSpell(target, SPELL_DOMINATION_CHAIN_PERIODIC, true);
+    }
+
+    void Register() override
+    {
+        OnEffectPeriodic += AuraEffectPeriodicFn(spell_sylvanas_windrunner_domination_chain_periodic::HandleEffectPeriodic, EFFECT_0, SPELL_AURA_PERIODIC_DAMAGE);
+    }
+};
+
+// 347609 - Wailing Arrow
+class spell_sylvanas_windrunner_wailing_arrow : public SpellScript
+{
+    PrepareSpellScript(spell_sylvanas_windrunner_wailing_arrow);
+
+    bool Validate(SpellInfo const* spellInfo) override
+    {
+        return ValidateSpellInfo({ spellInfo->GetEffect(EFFECT_0).TriggerSpell });
+    }
+
+    int32 CalcCastTime(int32 castTime) override
+    {
+        if (GetWorldStatePhase(GetCaster()) == PHASE_THREE)
+            return 1500;
+
+        return castTime;
+    }
+
+    void OnPrecast() override
+    {
+        Unit* caster = GetCaster();
+
+        if (GetWorldStatePhase(caster) == PHASE_ONE)
+        {
+            caster->m_Events.AddEvent(new SetSheatheOrNameplateOrAttackSpeed(caster, DATA_CHANGE_NAMEPLATE_TO_RIDING_COPY, 0), caster->m_Events.CalculateTime(1s + 531ms));
+            caster->m_Events.AddEvent(new SetSheatheOrNameplateOrAttackSpeed(caster, DATA_CHANGE_NAMEPLATE_TO_SYLVANAS, 0), caster->m_Events.CalculateTime(3s + 331ms));
+        }
+    }
+
+    void OnCast(SpellEffIndex /*effIndex*/)
+    {
+        Unit* caster = GetCaster();
+        if (!caster)
+            return;
+
+        caster->CastSpell(GetHitUnit(), GetEffectInfo(EFFECT_0).TriggerSpell, true);
```

Comment:
Check whether the target is null, if so, look for the highest threat to use as target.